pytential/symbolic/execution.py

```diff
@@ -281,10 +281,10 @@ def map_common_subexpression(self, expr):
             return self.rec(expr.child)
 
         try:
-            rec = cache[expr.child]
+            rec = thaw(cache[expr.child], self.array_context)
```

Comment:
There's little question that things in the discretization cache should be frozen. But the expr-scoped stuff... I think that's overkill.

Reply:
Great catch! Yes, fixed in adcaf93.